src/util/network.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_NETWORK_H
+#define BITCOIN_UTIL_NETWORK_H
+
+#include <string_view>
+
+enum class SocketAddrError {
+    OK, //!< No error
+    NO_HOST,
+    NO_PORT,
+    NO_HOSTPORT,
+};
+
+SocketAddrError hasValidHostPort(std::string_view socketAddr);
```

Comment:
I think there is no need to add a new file for this function. It can be put in `netbase.cpp`, looks in a similar category as e.g. `ParseNetwork()`.

Reply:
Agreed